subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/settings/views/ItrSettingViewModel.kt

```diff
@@ -63,6 +63,8 @@ class ItrSettingViewModel(
             subscriptions.getEntitlementStatus(ITR_PRODUCT_NAME).also {
                 if (it.isSuccess) {
                     _viewState.emit(viewState.value.copy(hasSubscription = it.getOrDefault(NotFound) == Found))
+                } else {
+                    _viewState.emit(viewState.value.copy(hasSubscription = false))
```

Comment:
Consider this scenario:
- User attempts to enable NetP (Connecting…)
- User returns to quickly Settings page
- Most likely the API calls would fail since we are changing the network tun.
- Netp successfully connects

we will get failure for the subscriptions.getEntitlementStatus call and we will hide the entry. We can only show it if the user goes in and out of the settings page again. Not 100% sure what the best approach here.

Reply:
Caught up on MM about it. This is a known issue for both ITR and PIR that will get solved in the next project when plugin in the correct screens (right now just shows a toast).